logger.py

```diff
@@ -45,30 +48,61 @@ def save_unpickled_test(data,protocol=pickle.DEFAULT_PROTOCOL):
         return
 
     #Writes the data to the files
-    with open(f'logs/protocol_{protocol}/unpickled/test_{testnumber}.txt', 'wb') as f:
+    with open(f'logs/{os_type}/{version_number}/protocol_{protocol}/test_{testnumber}.txt', 'wb') as f:
         f.write(hash_object.encode("utf-8"))
 
     #Writes the data to the json file
-    json.dump(data, open(f'logs/protocol_{protocol}/unpickled/test_{testnumber}.json', 'w'))
+    json.dump(data, open(f'logs/{os_type}/{version_number}/protocol_{protocol}/test_{testnumber}.json', 'w'))
 
-def unpack_and_commpare_pickle(testnumber=0,protocol=pickle.DEFAULT_PROTOCOL):
+def unpack_and_compare_single_test(testnumber=0,protocol=pickle.DEFAULT_PROTOCOL,os_type=False,version_number=False):
     '''Function used to unpack the pickle file and compare it to the json file. The function returns True if the files are the same'''
+    #If os is not given, we will use the running os and version
+    running_os_type,running_version_number = get_os_and_version()
+    if not os_type:
+        os_type = running_os_type
+    if not version_number:
+        version_number = running_version_number
 
-    with open(f'logs/protocol_{protocol}/unpickled/test_{testnumber}.txt', 'rb') as f:
+
+    with open(f'logs/{os_type}/{version_number}/protocol_{protocol}/test_{testnumber}.txt', 'rb') as f:
         hash_data_prev_py_version = f.read().decode("utf8")
 
     #Taking the data from the jason and transforming it to a new hash
-    json_data = json.load(open(f'logs/protocol_{protocol}/unpickled/test_{testnumber}.json', 'r'))
+    json_data = json.load(open(f'logs/{os_type}/{version_number}/protocol_{protocol}/test_{testnumber}.json', 'r'))
     pickle_object = pickle.dumps(json_data, protocol=protocol)
     hash_data_new_py_version = sha256(pickle_object).hexdigest()
 
     #Checking if the hashes are the same
     assert hash_data_prev_py_version == hash_data_new_py_version
 
-def clean_folder(foldername,protocol=pickle.DEFAULT_PROTOCOL):
+def get_os_and_version():
+    os_type = os.name
+    if os_type == "posix":
+        os_type = "linux"
+    elif os_type == "nt":
+        os_type = "windows"
+    else:
+        os_type = "macOS"
```

Comment:
This might be grabbed by the "posix" flag, as I believe it cover both. They are at least both UNIX, so worth making sure that this works.

Reply:
A fix has now been proposed.